src/calculation/MeasureReportBuilder.ts

```diff
@@ -393,6 +415,116 @@ export default class MeasureReportBuilder<T extends PopulationGroupResult> {
     }
   }
 
+  // Ratio requires different input types than other scores
+  private calcMeasureScoreRatio(detail: PopulationGroupResult, strataCode?: string) {
+    let observations = [];
+    if (detail.episodeResults) {
+      // find all episode results with a true measure observation
+      const relevantEpisodes =
+        detail.episodeResults?.filter(er => {
+          // ignore stratification (by setting as true) if stratifier not passed in
+          const inStrat = strataCode ? er.stratifierResults?.find(sr => sr.strataCode == strataCode)?.result : true;
+          return (
+            er.populationResults.filter(pr => pr.populationType === PopulationType.OBSERV)?.map(pr => pr.result) &&
+            inStrat
+          );
+        }) || [];
+      observations = relevantEpisodes.flatMap(er =>
+        er.populationResults
+          .filter(pr => pr.populationType === PopulationType.OBSERV)
+          .map(e => {
+            return { criteriaExpression: e.criteriaExpression, observation: e.observations ? e.observations[0] : 0 };
+          })
+      );
+    } else {
+      // ignore stratification (by setting as true) if stratifier not passed in
+      const inStrat = strataCode ? detail.stratifierResults?.find(sr => sr.strataCode === strataCode)?.result : true;
+
+      // Ratio for patient-based results is untested
+      const obsResultPop = detail.populationResults?.filter(pr => pr.populationType === PopulationType.OBSERV);
+      observations =
+        obsResultPop && inStrat
+          ? obsResultPop.map(e => {
+              return { criteriaExpression: e.criteriaExpression, observation: e.observations ? e.observations[0] : 0 };
+            })
+          : [];
+    }
+
+    // Calculate as (NUMER OBSERVATION) / (DENOM OBSERVATION)
+    const numeratorCount = observations
+      .filter(obs => obs.criteriaExpression === 'Numerator Observations')
+      ?.map(o => o.observation)
+      .reduce((accumulator, currentValue) => accumulator.observation + currentValue.observation);
+    const denominatorCount = observations
+      .filter(obs => obs.criteriaExpression === 'Denominator Observations')
+      ?.map(o => o.observation)
+      .reduce((accumulator, currentValue) => accumulator.observation + currentValue.observation);
```

Comment:
Discussed on a call, but should be respecting the `cqfm-aggregateMethod` extension defined on the populations